frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -17,6 +17,7 @@ import {
 
 type DropdownItemProps = {
   model: K8sKind;
+  resource: K8sResourceKind;
```

Comment:
I'm confused -- what am I missing here. This isn't used by the component, why did you add it to the props?

Reply:
@karthikjeeyar  ^^

Reply:
@andrewballantyne  When dropdownItem is selected, we are passing the selected resource (item from firehose) to the onChange handler, which takes K8sResourceKind as input. https://github.com/openshift/console/blob/977aca12752a297cac1124d0d106313ff569ab4b/frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelector.tsx#L22

Reply:
But the component that uses these props:

```
const DropdownItem: React.FC<DropdownItemProps> = ({ model, name }) => (
  <span className="co-resource-item">
    <span>
      <ResourceIcon kind={referenceForModel(model)} />
    </span>
    <span className="co-truncate show co-nowrap small">{name}</span>
  </span>
);
```
Doesn't use it... like at all. Why add it to the props? Firehose pushes a lot of props around into components that we don't use... is this somehow a TS error that `DropdownItem` doesn't define a prop?

Reply:
OnChange emitter from ResourceDropdown.tsx sends `key` and `value` (default value is  **string**) as arguments, In case of dropdownItem with badge the `value` argument is being set as a **React.Element**, so the newly introduced `resource` can be accessible as `ele?.props.resource` in the parent component. But I agree that the actual component does not require the `resource` at all . However I can add the key attribute
```
<span className="co-resource-item" key={resource.metadata.id} > 
....
</span>
``` 
let me know your thoughts on this

Reply:
Keys really only have 1 main purpose (and a few less common ones)... and that's for Array rendering of a list of elements. Using a key on a non list item has really only one valid purpose I can think of and that's to force a re-mount of a component. I do not believe I can see any benefit from having a key on a `span` that is not rendered as part of a `.map` (or equiv).

I do not believe there is any value in saying this component gets a prop it does not use. It's not really future proofing the code against accidental changes... if anything it's doing the opposite and coupling this component to a prop it does not use, reducing its reusability.

Reply:
Currently ResourceDropdown does not have a way to emit the selected resource to the parent component. I will add the logic to emit the selected resource on `onChange` event instead of having this resource prop tied up to the dropdownItem component.

Reply:
OH! NOW I understand what you're doing. Yeah... that's a very bad practice that you shouldn't do. Do not stuff variables into props of a component to extract them later from the internals of React. Components are not unbound objects you can just mutate to store data. 

Let me see what I can think of here.

Reply:
So yesterday night I tried to investigate sending the data in the onChange handler, but it's not available and a rework of the Dropdown would be needed.

I then back back to the drawing board and maybe tried to manually add the badge to the old implementation - but it seems the selection fails to go through -- I called it for a day there before figuring out why. I'll try to find some time to continue my investigation today but I have a very full plate, so no promises.